packages/core/src/render3/util/discovery_utils.ts

```diff
@@ -57,52 +46,29 @@ export function getComponent<T = {}>(element: Element): T|null {
 }
 
 /**
- * Returns the component instance associated with a given DOM host element.
- * Elements which don't represent components return `null`.
- *
- * @param element Host DOM element from which the component should be retrieved.
- *
- * ```
- * <my-app>
- *   #VIEW
- *     <div>
- *       <child-comp></child-comp>
- *     </div>
- * </mp-app>
+ * Retrieves the instance of the component that an element is inside of.
```

Comment:
What's the difference between this function and `getViewComponent`?  I think this needs an example that shows it.

Reply:
I wanted to raise this in the next meeting, because I'm not exactly sure what the difference is either. From my testing the return values were identical.

Reply:
@mhevery ?

Reply:
Talked to Misko offline about this and he reminded me of embedded views. This will return the context of the closest embedded view if the element is inside an embedded view. So maybe:

```suggestion
 * If inside an embedded view (e.g. *ngIf or *ngFor), retrieves the context of the embedded view that owns the element.
 * If not, retrieves the instance of the component that owns the element (in this case, the result should be the same as `getOwnerComponent`

```